tests/qtrajectory_test.cc

```diff
@@ -132,8 +226,9 @@ void AddGenAmplDumpNoise(
        {normal, 0, p3, {M::Create(time, 1, {0, 0, 0, 0, s2, 0, 0, 0})}}});
 }
 
-template <typename AddNoise>
-types::NoisyCircuit GenerateNoisyCircuit(double p, AddNoise&& add_noise) {
+template <typename AddNoise1, typename AddNoise2>
+types::NoisyCircuit GenerateNoisyCircuit(
```

Comment:
My read is that the core behavior of this test isn't changing, you've just updated it to take advantage of the looser time-order restriction. Is that correct?

Reply:
Yes, that is correct. This basically shows an alternative way of constructing Kraus operators.